server/src/main/java/io/deephaven/server/session/SharedTicketResolver.java

```diff
@@ -131,88 +141,77 @@ public <T> SessionState.ExportBuilder<T> publish(
+                            + "' to shared ticket '" + ticketHex + "' (hex): destination already exists"),
                     null);
+        } else if (onPublish != null) {
+            onPublish.run();
         }
     }
 
     /**
-     * Convenience method to convert from a shared variable identifier to Flight.Ticket
+     * Convenience method to convert from a shared identifier to Flight.Ticket
      *
-     * @param identifier the shared variable identifier to convert
+     * @param identifier the shared identifier to convert
      * @return the flight ticket this descriptor represents
      */
-    public static Flight.Ticket flightTicketForId(final ByteString identifier) {
+    public static Flight.Ticket flightTicketForId(final byte[] identifier) {
         return Flight.Ticket.newBuilder()
-                .setTicket(identifier)
+                .setTicket(ByteString.copyFrom(identifier))
                 .build();
     }
 
     /**
-     * Convenience method to convert from a shared variable identifier to Ticket
+     * Convenience method to convert from a shared identifier to Ticket
      *
-     * @param identifier the shared variable identifier to convert
+     * @param identifier the shared identifier to convert
      * @return the flight ticket this descriptor represents
      */
-    public static io.deephaven.proto.backplane.grpc.Ticket ticketForId(final ByteString identifier) {
+    public static io.deephaven.proto.backplane.grpc.Ticket ticketForId(final byte[] identifier) {
         return io.deephaven.proto.backplane.grpc.Ticket.newBuilder()
-                .setTicket(identifier)
+                .setTicket(ByteString.copyFrom(identifier))
                 .build();
     }
 
     /**
-     * Convenience method to convert from a shared variable identifier to Flight.FlightDescriptor
+     * Convenience method to convert from a shared identifier to Flight.FlightDescriptor
      *
-     * @param identifier the shared variable identifier to convert
+     * @param identifier the shared identifier to convert
      * @return the flight descriptor this descriptor represents
      */
-    public static Flight.FlightDescriptor descriptorForId(final ByteString identifier) {
+    public static Flight.FlightDescriptor descriptorForId(final byte[] identifier) {
         return Flight.FlightDescriptor.newBuilder()
                 .setType(Flight.FlightDescriptor.DescriptorType.PATH)
-                .addAllPath(SharedTicketHelper.nameToPath(identifier))
+                .addAllPath(SharedTicketHelper.idToPath(identifier))
                 .build();
     }
 
     /**
-     * Convenience method to convert from a Flight.Ticket (as ByteBuffer) to shared variable identifier
+     * Convenience method to convert from a Flight.Ticket (as ByteBuffer) to shared identifier
      *
      * @param ticket the ticket to convert
      * @param logId an end-user friendly identification of the ticket should an error occur
```

Comment:
We have public `ByteString` methods here. Is that intentional? I know it's "allowed" on the server side